stl/inc/format

```diff
@@ -2782,98 +2783,103 @@ _NODISCARD auto make_wformat_args(const _Args&... _Vals) {
     return _Format_arg_store<_Context, _Args...>{_Vals...};
 }
 
-template <class _Out, class _CharT>
-using format_args_t = basic_format_args<basic_format_context<_Out, _CharT>>;
-
 template <output_iterator<const char&> _OutputIt>
-_OutputIt vformat_to(
-    _OutputIt _Out, const string_view _Fmt, const format_args_t<type_identity_t<_OutputIt>, char> _Args) {
-    _Format_handler<_OutputIt, char, basic_format_context<_OutputIt, char>> _Handler(
-        _STD move(_Out), _Fmt, _Args, locale::classic());
-    _Parse_format_string(_Fmt, _Handler);
-    return _Handler._Ctx.out();
+_OutputIt vformat_to(_OutputIt _Out, const string_view _Fmt, const format_args _Args) {
+    if constexpr (std::is_same_v<_OutputIt, _Fmt_it>) {
+        _Format_handler<_Fmt_it, char> _Handler(_Out, _Fmt, _Args, locale::classic());
+        _Parse_format_string(_Fmt, _Handler);
+        return _Out;
+    } else {
+        _Fmt_iterator_buffer<_OutputIt, char> _Buf(_STD move(_Out));
+        _Format_handler<_Fmt_it, char> _Handler(_Fmt_it{_Buf}, _Fmt, _Args, locale::classic());

```

Comment:
perhaps a note on `_Format_handler` would be useful, indicating that `_OutputIt` is always `_Fmt_it` or `_Fmt_wit`

Reply:
Actually it doesn't make sense to parameterize `_Format_handler` on output iterator at all. Removed parameterization and simplified the whole thing even more.